Vwitter/View Controllers/SearchUsersViewController.m

```diff
@@ -0,0 +1,92 @@
+//
+//  SearchUsersViewController.m
+//  Vwitter
+//
+//  Created by Christina Li on 7/8/22.
+//
+
+#import "SearchUsersViewController.h"
+#import "UserCell.h"
+#import <Parse/Parse.h>
+
+@interface SearchUsersViewController () <UITableViewDataSource, UITableViewDelegate, UISearchBarDelegate>
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+
+@property (strong, nonatomic) NSMutableArray *arrayOfUsers;
+@property (strong, nonatomic) NSArray *filteredData;
+@property (weak, nonatomic) IBOutlet UISearchBar *searchBar;
+// make a property that stores the people that the user is following
+
+@end
+
+@implementation SearchUsersViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    [self loadData];
+    
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.searchBar.delegate = self;
+    self.filteredData = self.arrayOfUsers;
+    
+}
+
+- (void)loadData {
+    PFQuery *userQuery = [PFUser query];
+    userQuery.limit = 20;
+    __weak typeof(self) weakSelf = self;
+    [userQuery findObjectsInBackgroundWithBlock:^(NSArray<PFUser *> * _Nullable users, NSError * _Nullable error) {
+        typeof(self) strongSelf = weakSelf;
+        if (!strongSelf) {
+            NSLog(@"I got killed!");
+            return;
+        }
+        if (users) {
+            NSLog(@"😎😎😎 Successfully loaded search users timeline");
+            strongSelf.arrayOfUsers = users.mutableCopy;
+            strongSelf.filteredData = strongSelf.arrayOfUsers;
+            [strongSelf.tableView reloadData];
+            
+        }
+        else {
+            NSLog(@"😫😫😫 Error getting home timeline: %@", error.localizedDescription);
+        }
+    }];
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
+    return self.filteredData.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UserCell *cell = [tableView dequeueReusableCellWithIdentifier:@"UserCell" forIndexPath:indexPath];
+//    if this userid is in the followingSet, then do stuff inside the usercell
+
+    cell.user = self.filteredData[indexPath.row];
+
+    return cell;
```

Comment:
Eventually you want to make it so that both users _and_ vents can show up in the search results right?

Once you have defined the cloud code function that returns both kinds of objects in an array, you would change your data array to be something like `NSMutableArray<PFObject *>`. Then in the `cellForRowAtIndexPath` method you could use `[searchResults[indexPath.row] isKindOfClass:VWUser.class]` to check if it's a user (and if so, dequeue a UserCell, and if it's a VWVent class, return some new `VWVentSearchCell` that you would create and register with the table view). 

You could also just write more code in the UserCell so that it could also display a vent, but it is best to do it as a separate UITableViewCell subclass to keep things simple.